File: core/src/SSRAG/Models/FlowVariable.cs
using SSRAG.Datory;
using SSRAG.Datory.Annotations;
using SSRAG.Enums;

namespace SSRAG.Models
{
    [DataTable("ssrag_flow_variable")]
    public class FlowVariable : Entity
    {
        [DataColumn]
        public int SiteId { get; set; }

        [DataColumn]
        public string NodeId { get; set; }

        [DataColumn]
        public VariableType Type { get; set; }

        [DataColumn]
        public string Name { get; set; }

        [DataColumn]
        public VariableDataType DataType { get; set; }

        [DataColumn]
        public string Description { get; set; }

        [DataColumn]
        public bool IsDisabled { get; set; }

        [DataColumn]
        public bool IsReference { get; set; }

        [DataColumn]
        public string ReferenceNodeId { get; set; }

        [DataColumn]
        public string ReferenceName { get; set; }

        [DataColumn]
        public string Value { get; set; }
    }
}

File: web/src/models/flowVariable.ts
import { VariableDataType } from '../enums/variableDataType'
import { VariableType } from '../enums/variableType'

export interface FlowVariable {
  id?: number
  nodeId?: string
  type?: VariableType
  key?: string
  name?: string
  dataType?: VariableDataType
  value?: Record<string, unknown> | string | number
  isDisabled?: boolean
  isReference?: boolean
  referenceNodeId?: string
  referenceName?: string
  description?: string
}
